micrometer-spring-legacy/src/samples/resources/application-prometheus.yml

```diff
@@ -14,4 +14,12 @@
 # limitations under the License.
 #
 
-spring.metrics.prometheus.enabled: true
\ No newline at end of file
+management.security.enabled: false
+
+spring.metrics:
+  prometheus.enabled: true
+  overrides:
+    ALL.enabled: false
+    jvm.enabled: true
+    jvm.memory.used: disabled
```

Comment:
@jkschneider I'm liking the value `overrides` to represent the configuration. You can see the example above.

I also added in config validation/error propagation. Let me know if there are any other needs.